src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/NomisCommentService.kt

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.entity.CalculationRequest
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.enumerations.ReleaseDateType
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.ManualEntrySelectedDate
+import java.time.LocalDate
+
+@Service
+class NomisCommentService {
+
+  fun getNomisComment(
+    calculationRequest: CalculationRequest,
+    isSpecialistSupport: Boolean,
+    approvedDates: List<ManualEntrySelectedDate>?,
+  ): String {
+    val comment = if (isSpecialistSupport) {
+      SPECIALIST_SUPPORT_COMMENT
+    } else if (approvedDates?.isNotEmpty() == true) {
+      MANUAL_ENTERED_DATES_COMMENT
+    } else if (!calculationRequest.reasonForCalculation?.isOther!!) {
+      DEFAULT_COMMENT
+    } else {
+      OTHER_COMMENT
+    }
+
+    return if (comment == OTHER_COMMENT) {
+      comment.format(calculationRequest.calculationReference)
+    } else {
+      comment.format(
+        calculationRequest.reasonForCalculation?.displayName,
+        calculationRequest.calculationReference,
+      )
+    }
+  }
+
+  fun getManualNomisComment(
+    calculationRequest: CalculationRequest,
+    dates: Map<ReleaseDateType, LocalDate?>,
+    isGenuineOverride: Boolean,
+  ): String {
+    var comment = if (dates.containsKey(ReleaseDateType.None)) INDETERMINATE_COMMENT else MANUAL_ENTRY_COMMENT
```

Comment:
I think this is unnecessary reassignment, this could be:
```
var comment = if (isGenuineOverride) MANUALLY_ENTERED_OVERRIDE 
else if (dates.containsKey(ReleaseDateType.None)) INDETERMINATE_COMMENT 
else MANUAL_ENTRY_COMMENT
```